string source_robot
string target_robot
string message